mycore-base/src/main/java/org/mycore/datamodel/classifications2/impl/MCREventedCategoryDAOImpl.java

```diff
@@ -0,0 +1,127 @@
+        queueForCommit(evt);
+        return rv;
+    }
+
+    @Override
+    public void deleteCategory(MCRCategoryID id) {
+        MCRCategory category = super.getCategory(id, -1);
+        if (category == null) {
+            throw new MCRPersistenceException("Category " + id + " was not found. Delete aborted.");
+        }
+        MCREvent evt = new MCREvent(MCREvent.CLASS_TYPE, MCREvent.DELETE_EVENT);
+        evt.put("class", category);
+        manager.handleEvent(evt, MCREventManager.BACKWARD);
+        queueForCommit(evt);
+        super.deleteCategory(id);
+    }
+
+    @Override
+    public void moveCategory(MCRCategoryID id, MCRCategoryID newParentID, int index) {
+        MCREvent evt = new MCREvent(EVENT_OBJECT, MCREvent.UPDATE_EVENT);
+        evt.put("class", super.getCategory(id, -1));
+        evt.put("parent", super.getCategory(newParentID, -1));
+        evt.put("index", index);
+        // Type is used for specifying a special Update operation
+        // originally named UType (Update Type), it is an Optional Value
+        evt.put("type", "move");
+        manager.handleEvent(evt);
+        queueForCommit(evt);
+        super.moveCategory(id, newParentID, index);
+    }
+
+    @Override
+    public MCRCategory removeLabel(MCRCategoryID id, String lang) {
+        MCRCategory rv = super.removeLabel(id, lang);
+        MCREvent evt = new MCREvent(EVENT_OBJECT, MCREvent.UPDATE_EVENT);
+        evt.put("class", super.getCategory(id, -1));
+        manager.handleEvent(evt);
+        queueForCommit(evt);
+        return rv;
+    }
+
+    @Override
+    public Collection<MCRCategoryImpl> replaceCategory(MCRCategory newCategory) throws IllegalArgumentException {
+        Collection<MCRCategoryImpl> rv = super.replaceCategory(newCategory);
+        MCREvent evt = new MCREvent(EVENT_OBJECT, MCREvent.UPDATE_EVENT);
+        evt.put("class", newCategory);
+        manager.handleEvent(evt);
+        queueForCommit(evt);
+        return rv;
+    }
+
+    @Override
+    public MCRCategory setLabel(MCRCategoryID id, MCRLabel label) {
+        MCRCategory rv = super.setLabel(id, label);
+        MCREvent evt = new MCREvent(EVENT_OBJECT, MCREvent.UPDATE_EVENT);
+        evt.put("class", super.getCategory(id, -1));
+        manager.handleEvent(evt);
+        queueForCommit(evt);
+        return rv;
+    }
+
+    @Override
+    public MCRCategory setLabels(MCRCategoryID id, Set<MCRLabel> labels) {
+        MCRCategory rv = super.setLabels(id, labels);
+        MCREvent evt = new MCREvent(EVENT_OBJECT, MCREvent.UPDATE_EVENT);
+        evt.put("class", super.getCategory(id, -1));
+        manager.handleEvent(evt);
+        queueForCommit(evt);
+        return rv;
+    }
+
+    @SuppressWarnings("unchecked")
+    protected void queueForCommit(MCREvent evt) {
+        String classQueue = "classQueue";
+        MCRSession currentSession = MCRSessionMgr.getCurrentSession();
+        ((ArrayList<MCREvent>) currentSession.get(classQueue)).add(evt);
```

Comment:
String "classQueue" is used to often in Code. This should be centralized somewhere.
Also a BIG no go is to use session as a storage for transactional data without any transactional reference. There could be two different transactions running concurrently and one is rolled back. In this case all the data of two transactions are lost.